src/components/files-grid-viewer.js

```diff
@@ -1,69 +1,59 @@
+import React from 'react';
 import { Tooltip } from 'antd';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
-import React, { useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { RESOURCE_TYPE } from '../domain/constants.js';
 import DeleteIcon from './icons/general/delete-icon.js';
 import PreviewIcon from './icons/general/preview-icon.js';
+import { cdnObjectShape } from '../ui/default-prop-types.js';
 import { confirmCdnFileDelete } from './confirmation-dialogs.js';
+import { CDN_OBJECT_TYPE, RESOURCE_TYPE } from '../domain/constants.js';
 import { getResourceIcon, getResourceType } from '../utils/resource-utils.js';
 import { FolderFilledIconComponent } from './icons/files/folder-filled-icon.js';
 
 function FilesGridViewer({
   files,
+  selectedFileUrl,
   canDelete,
   canNavigateToParent,
   onDeleteClick,
+  onFileClick,
   onPreviewClick,
-  onSelectionChange,
   onNavigateToParentClick
 }) {
   const { t } = useTranslation('filesGridViewer');
-  const [selectedFile, setSelectedFile] = useState(null);
-
-  const handleFileClick = file => {
-    const newSelectedFile = file.name === selectedFile?.name || selectedFile?.isDirectory ? null : file;
-
-    setSelectedFile(newSelectedFile);
-    onSelectionChange(newSelectedFile);
-  };
 
   const handlePreviewClick = (event, file) => {
     event.stopPropagation();
     onPreviewClick(file);
   };
 
-  const handleDeleteFile = file => {
-    onDeleteClick(file);
-  };
-
   const handleDeleteClick = (event, file) => {
     event.stopPropagation();
-    confirmCdnFileDelete(t, file.name, () => handleDeleteFile(file));
+    confirmCdnFileDelete(t, file.displayName, () => onDeleteClick(file));
   };
 
   const renderFile = file => {
     let fileDisplay;
     if (getResourceType(file.url) === RESOURCE_TYPE.image) {
       fileDisplay = <img className="FilesGridViewer-fileDisplayImage" src={file.url} />;
     } else {
-      const Icon = getResourceIcon({ filePath: file.path, isDirectory: file.isDirectory, filled: true });
+      const Icon = getResourceIcon({ url: file.url, isDirectory: file.type === CDN_OBJECT_TYPE.directory, filled: true });
       fileDisplay = <Icon />;
     }
-    const overlayClasses = classNames('FilesGridViewer-fileOverlay', { 'is-visible': file.name === selectedFile?.name });
-    const actionsClasses = classNames('FilesGridViewer-actions', { 'are-visible': file.name === selectedFile?.name });
+    const overlayClasses = classNames('FilesGridViewer-fileOverlay', { 'is-visible': file.portableUrl === selectedFileUrl });
+    const actionsClasses = classNames('FilesGridViewer-actions', { 'are-visible': file.portableUrl === selectedFileUrl });
 
     return (
-      <div className="FilesGridViewer-fileContainer" key={file.path}>
-        <a className="FilesGridViewer-file" onClick={() => handleFileClick(file)}>
+      <div className="FilesGridViewer-fileContainer" key={file.portableUrl}>
```

Comment:
Nice 👍  With this `portableUrl` we're one step closer to being able to remove the `sourceType` toggle, since we'll be able to deduce the source type wherever needed.